crates/ruff_linter/src/rules/pylint/rules/repeated_equality_comparison.rs

```diff
@@ -72,79 +70,85 @@ impl AlwaysFixableViolation for RepeatedEqualityComparison {
-    let mut value_to_comparators: FxHashMap<HashableExpr, (TextSize, Vec<&Expr>, Vec<&Expr>)> =
+    let mut value_to_comparators: FxHashMap<HashableExpr, (TextSize, Vec<&Expr>, Vec<usize>)> =
         FxHashMap::with_capacity_and_hasher(bool_op.values.len() * 2, FxBuildHasher);
 
-    for value in &bool_op.values {
-        // Enforced via `is_allowed_value`.
-        let Expr::Compare(ast::ExprCompare {
-            left, comparators, ..
-        }) = value
-        else {
-            return;
-        };
-
-        // Enforced via `is_allowed_value`.
-        let [right] = &**comparators else {
-            return;
+    for (i, value) in bool_op.values.iter().enumerate() {
+        let Some((left, right)) = to_allowed_value(bool_op.op, value, checker.semantic()) else {
+            continue;
         };
 
-        if matches!(left.as_ref(), Expr::Name(_) | Expr::Attribute(_)) {
-            let (_, left_matches, value_matches) = value_to_comparators
-                .entry(left.deref().into())
+        if matches!(left, Expr::Name(_) | Expr::Attribute(_)) {
+            let (_, left_matches, index_matches) = value_to_comparators
+                .entry(left.into())
                 .or_insert_with(|| (left.start(), Vec::new(), Vec::new()));
             left_matches.push(right);
-            value_matches.push(value);
+            index_matches.push(i);
         }
 
         if matches!(right, Expr::Name(_) | Expr::Attribute(_)) {
-            let (_, right_matches, value_matches) = value_to_comparators
+            let (_, right_matches, index_matches) = value_to_comparators
                 .entry(right.into())
                 .or_insert_with(|| (right.start(), Vec::new(), Vec::new()));
             right_matches.push(left);
-            value_matches.push(value);
+            index_matches.push(i);
         }
     }
 
-    for (value, (start, comparators, values)) in value_to_comparators
+    for (value, (_, comparators, indices)) in value_to_comparators
         .iter()
         .sorted_by_key(|(_, (start, _, _))| *start)
     {
-        if comparators.len() > 1 {
+        // If there's only one comparison, there's nothing to merge.
+        if comparators.len() == 1 {
+            continue;
+        }
+
+        // Break into sequences of consecutive comparisons.
+        let mut sequences: Vec<(Vec<usize>, Vec<&Expr>)> = Vec::new();
+        let mut last = None;
+        for (index, comparator) in indices.iter().zip(comparators.iter()) {
+            if let Some(last) = last {
+                if last + 1 == *index {
+                    let (indices, comparators) = sequences.last_mut().unwrap();
+                    indices.push(*index);
+                    comparators.push(*comparator);
+                    continue;
+                }
+            }
+            sequences.push((vec![*index], vec![*comparator]));
+            last = Some(*index);
+        }
```

Comment:
The value of `last` needs to be updated before the `continue` in the inner condition to avoid the bug as mentioned in my other comment.

Reply:
```diff
diff --git a/crates/ruff_linter/src/rules/pylint/rules/repeated_equality_comparison.rs b/crates/ruff_linter/src/rules/pylint/rules/repeated_equalit>
index db2d9d7e5..adb9544c3 100644
--- a/crates/ruff_linter/src/rules/pylint/rules/repeated_equality_comparison.rs
+++ b/crates/ruff_linter/src/rules/pylint/rules/repeated_equality_comparison.rs
@@ -109,15 +109,13 @@ pub(crate) fn repeated_equality_comparison(checker: &mut Checker, bool_op: &ast:
         let mut sequences: Vec<(Vec<usize>, Vec<&Expr>)> = Vec::new();
         let mut last = None;
         for (index, comparator) in indices.iter().zip(comparators.iter()) {
-            if let Some(last) = last {
-                if last + 1 == *index {
-                    let (indices, comparators) = sequences.last_mut().unwrap();
-                    indices.push(*index);
-                    comparators.push(*comparator);
-                    continue;
-                }
+            if last.is_some_and(|last| last + 1 == *index) {
+                let (indices, comparators) = sequences.last_mut().unwrap();
+                indices.push(*index);
+                comparators.push(*comparator);
+            } else {
+                sequences.push((vec![*index], vec![*comparator]));
             }
-            sequences.push((vec![*index], vec![*comparator]));
             last = Some(*index);
         }
```